src/Sign.Core/DataFormatSigners/ClickOnceSigner.cs

```diff
@@ -273,5 +281,60 @@ public void CopySigningDependencies(FileInfo deploymentManifestFile, DirectoryIn
                 }
             }
         }
+
+        /// <summary>
+        /// Try and find the application manifest (.manifest) file from a ClickOnce application manifest (.application / .vsto
+        /// There might not be one, if the user is attempting to only re-sign the deployment manifest without touching other files.
+        /// This is necessary because there might be multiple *.manifest files present, e.g. if a DLL that's part of the ClickOnce
+        /// package ships its own assembly manifest which isn't a ClickOnce application manifest.
+        /// </summary>
+        /// <param name="deploymentManifest">A <see cref="FileInfo"/> representing a deployment manifest file.</param>
+        /// <param name="applicationManifestFileName">A <see cref="string?"/> representing a manifest file name or <c>null</c> if one isn't found.</param>
+        /// <returns><c>true</c> if the application manifest file name was found; otherwise, <c>false</c>.</returns>
+        /// <remarks>This is non-private only for unit testing.</remarks>
+        internal bool TryGetApplicationManifestFileName(FileInfo deploymentManifest, [NotNullWhen(true)] out string? applicationManifestFileName)
+        {
+            applicationManifestFileName = null;
+
+            XmlDocument xmlDoc = _xmlDocumentLoader.Load(deploymentManifest);
+
+            // there should only be a single result here, if the file is a valid clickonce manifest.
+            XmlNodeList dependentAssemblies = xmlDoc.GetElementsByTagName("dependentAssembly");
+            if (dependentAssemblies.Count != 1)
+            {
+                Logger.LogDebug(Resources.ApplicationManifestNotFound);
+                return false;
+            }
+
+            XmlNode? node = dependentAssemblies.Item(0);
+            if (node is null || node.Attributes is null)
+            {
+                Logger.LogDebug(Resources.ApplicationManifestNotFound);
+                return false;
+            }
```

Comment:
There seems to be a bug here.  The comment asserts that there should only be one dependentAssembly element.  However, it seems that there can be multiple.

From https://learn.microsoft.com/visualstudio/deployment/dependency-element-clickonce-deployment?view=vs-2022#elements-and-attributes:

> The `dependency` element is required. It has no attributes. A deployment manifest can have multiple `dependency` elements.

This means this change would break users with multiple `dependency` elements.  This method needs smarter application manifest identification.